test/modules/survey/components/Survey.test.js

```diff
@@ -0,0 +1,153 @@
+		const { embed = false } = config;
+
+		const initialState = {
+			mainMenu: {
+				open: true
+			},
+			notifications: {
+				latest: null
+			},
+			media: {
+				portrait: false,
+				minWidth: true
+			},
+			...state
+		};
+
+		store = TestUtils.setupStoreAndDi(initialState, { mainMenu: createNoInitialStateMainMenuReducer(), media: createNoInitialStateMediaReducer(), notifications: notificationReducer });
+		$injector.registerSingleton('EnvironmentService', {
+			isEmbedded: () => embed
+		});
+		$injector.registerSingleton('TranslationService', { translate: (key) => key });
+
+		return TestUtils.render(Survey.tag);
+	};
+
+	describe('when initialized', () => {
+		beforeEach(async () => {
+			jasmine.clock().install();
+		});
+
+		afterEach(function () {
+			jasmine.clock().uninstall();
+		});
+
+		it('adds survey for landscape desktop', async () => {
+			const state = {
+				media: {
+					portrait: false,
+					minWidth: true
+				}
+			};
+			const element = await setup(state, {});
+			expect(element.shadowRoot.querySelectorAll('.is-landscape')).toHaveSize(1);
+			expect(element.shadowRoot.querySelectorAll('.is-desktop')).toHaveSize(1);
+			expect(element.shadowRoot.querySelectorAll('.survey__button')).toHaveSize(1);
+			expect(window.getComputedStyle(element.shadowRoot.querySelector('.survey__button')).display).toBe('flex');
+		});
+
+		it('adds survey for landscape tablet', async () => {
+			const state = {
+				media: {
+					portrait: false,
+					minWidth: false
+				}
+			};
+			const element = await setup(state, {});
+			expect(element.shadowRoot.querySelectorAll('.is-landscape')).toHaveSize(1);
+			expect(element.shadowRoot.querySelectorAll('.is-tablet')).toHaveSize(1);
+			expect(element.shadowRoot.querySelectorAll('.survey__button')).toHaveSize(1);
+			expect(window.getComputedStyle(element.shadowRoot.querySelector('.survey__button')).display).toBe('flex');
+		});
+
+		it('adds survey for portrait mode', async () => {
+			const state = {
+				media: {
+					portrait: true
+				}
+			};
+			const element = await setup(state, {});
+			expect(element.shadowRoot.querySelectorAll('.survey__button')).toHaveSize(1);
+			expect(window.getComputedStyle(element.shadowRoot.querySelector('.survey__button')).display).toBe('none');
+		});
+
+		it('emits a notification', async () => {
+			const state = {
+				media: {
+					portrait: true
+				}
+			};
+
+			const element = await setup(state, {});
+			expect(store.getState().notifications.latest).toBeNull();
+			jasmine.clock().tick(3100);
+
+			expect(element).toBeTruthy();
+			expect(isTemplateResult(store.getState().notifications.latest.payload.content)).toBeTrue();
+		});
+
+
+		it('emits a notification with a close-Button and a open-Button', async () => {
+			const target = document.createElement('div');
+			const state = {
+				media: {
+					portrait: true
+				}
+			};
+
+			const element = await setup(state, {});
+			jasmine.clock().tick(3100);
```

Comment:
We should export the delay constant in `Survey.js`, import and use it here: 
`jasmine.clock().tick(Survey_Notification_Delay_Time+100);`

In Survey.js we can also test the value of the constant then.